testing/test_gateway.py

```diff
@@ -122,6 +122,37 @@ def test_remote_exec_module(self, tmpdir, gw):
         name = channel.receive()
         assert name == 2
 
+    def test_remote_exec_module_is_removed(self, gw, tmpdir, monkeypatch):
+        remotetest = tmpdir.join("remote.py")
+        remotetest.write(
+            dedent(
+                """
+            def remote():
+                return True
+
+            if __name__ == '__channelexec__':
+                for item in channel:  # noqa
+                    channel.send(eval(item))  # noqa
+
+            """
+            )
+        )
+
+        monkeypatch.syspath_prepend(tmpdir)
+        import remote
+
+        ch = gw.remote_exec(remote)
+        # simulate sending the code to a remote location that does not have
+        # access to the source
+        tmpdir.remove()
```

Comment:
So, the right order should be:
```
import remote
linecache.updatecache(...)
tmpdir.remove()
gw.remote_exec(...)
```
so when remote_exec() is is ensured that the file on disk is already gone, but its contents are cached in memory so they are pickled and sent.

Reply:
Oh my apologies, I notice now that linecache.updatecache throws away the contents if the file is not found on disk. You are correct.